mautic/mautic
# populate doNotContact field in webhooks

| Q  | A
| --- | ---
| Bug fix? | Yes |
| New feature? | No |
| Automated tests included? | No |
| Related user documentation PR URL | / |
| Related developer documentation PR URL | / |
| Issues addressed (#s or URLs) | / |
| BC breaks? | No |
| Deprecations? | No |

#### Description:
doNotContact elements in webhooks are empty. This PR, properly populate this field.


#### Steps to reproduce the bug:
1 Go to [https://webhook.site/](https://webhook.site/) and grab the unique URL
2. Create webhook in Mautic and check "Contact Updated Event", "Contact subscription change", "Contact point change" and "Contact deleted"
3. Trigger each of those webhooks for a contact in the DNC list (eg update the contact, change their segments, change points, delete contact)
 3. Check webhook's payload on  [https://webhook.site/](https://webhook.site/), notice that the doNotContact elements are empty:
```
    "doNotContact": [
        []
    ],
```
#### Steps to test this PR:
1 Go to [https://webhook.site/](https://webhook.site/) and grab the unique URL
2. Create webhook in Mautic and check "Contact Updated Event", "Contact subscription change", "Contact point change" and "Contact deleted"
3. Trigger each of those webhooks for a contact in the DNC list (eg update the contact, change their segments, change points, delete contact)
 3. Check webhook's payload on  [https://webhook.site/](https://webhook.site/), notice that the doNotContact elements no longer are empty:
```
        "doNotContact": [
          {
            "id": 1,
            "reason": 0,
            "comments": "",
            "channel": "email",
            "channelId": null
          }
        ],
```
